[12.0] ADD attachment_db_by_checksum

eLBati opened the pull request:
DB attachments saved by checksum

Allow to identify database attachments through their hash, avoiding duplicates

eLBati committed:
ADD attachment_db_by_checksum

eLBati commented:
Does anybody know what's happening to travis?

simahawk commented on attachment_db_by_checksum/models/ir_attachment.py:
```diff
@@ -0,0 +1,72 @@
+            return super(Attachment, self)._file_write(value, checksum)
+        fname, _ = self._get_path(False, checksum)
+        att = self.env["ir.attachment.content"].search([
+            ("checksum", "=", fname)
```
```suggestion
            ("checksum", "=", fname), limit=1
```
same for all similar search (which could be delegated to `_find_attachment_content`

eLBati replied:
Done

simahawk commented on attachment_db_by_checksum/models/ir_attachment.py:
```diff
@@ -0,0 +1,72 @@
+            ("checksum", "=", checksum)
+        ])
+        if not att:
+            _logger.info("File %s not found" % checksum)
```
```suggestion
            _logger.debug("File %s not found" % checksum)
```

eLBati replied:
Done

eLBati committed:
FIX attachment_db_by_checksum
 - while deleting attachment content, search all the attachments, otherwise we delete content still used
 - checksum is longer than 40; update its content

simahawk reviewed (commented):
try rebasing to fix the build

eLBati committed:
pre-commit

eLBati commented:
@simahawk done thanks
